argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,274 @@
+
+        # Find the indices of the "first" detection assigned to each GT.
+        unique_gt_matches, unique_dt_matches = np.unique(gt_matches, return_index=True)
+        for i, thr in enumerate(self.dt_cfg.sim_ths):
+
+            # tp_mask may need to be defined differently with other similarity metrics
+            tp_mask = match_scores[unique_dt_matches] > -thr
+            error_types[unique_dt_matches, i] = tp_mask
+
+            if thr == self.dt_cfg.tp_thresh and np.count_nonzero(tp_mask) > 0:
+                dt_tp_indices = unique_dt_matches[tp_mask]
+                gt_tp_indices = unique_gt_matches[tp_mask]
+
+                dt_df = pd.DataFrame([dt.__dict__ for dt in dts[ranks][dt_tp_indices]])
+                gt_df = pd.DataFrame([gt.__dict__ for gt in gts[gt_tp_indices]])
+
+                trans_error = dist_fn(dt_df, gt_df, DistFnType.TRANSLATION)
+                scale_error = dist_fn(dt_df, gt_df, DistFnType.SCALE)
+                orient_error = dist_fn(dt_df, gt_df, DistFnType.ORIENTATION)
+
+                error_types[dt_tp_indices, n_threshs : n_threshs + 3] = np.vstack(
+                    (trans_error, scale_error, orient_error)
+                ).T
+
+        return np.hstack((error_types, scores))
+
+    def summarize(self, data: defaultdict, cls_inst_map: defaultdict) -> defaultdict:
+        """Calculate and print the detection metrics.
+
+        Args:
+            data: The aggregated data used for summarization.
+            cls_inst_map: Map of classes to number of instances.
+
+        Returns:
+            summary: The summary statistics.
+        """
+        summary = defaultdict(list)
+        rec_interp = np.linspace(0, 1, self.dt_cfg.n_rec_samples)
+        num_ths = len(self.dt_cfg.sim_ths)
+        if not self.fig_fpath.is_dir():
+            self.fig_fpath.mkdir(parents=True, exist_ok=True)
+        for cls_name, cls_stats in data.items():
+            num_inst = cls_inst_map[cls_name]
+            ranks = cls_stats[:, -1].argsort()[::-1]
+            cls_stats = cls_stats[ranks]
+            for i, _ in enumerate(self.dt_cfg.sim_ths):
+                tp = cls_stats[:, i].astype(bool)
+
+                cumulative_tp = np.cumsum(tp, dtype=np.int)
+                cumulative_fp = np.cumsum(~tp, dtype=np.int)
+                cumulative_fn = num_inst - cumulative_tp
+
+                precs = cumulative_tp / (cumulative_tp + cumulative_fp + np.finfo(float).eps)
+                recs = cumulative_tp / (cumulative_tp + cumulative_fn)
+
+                precs = interp(precs)
+                prec_interp = np.interp(rec_interp, recs, precs, right=0)
+                prec_trun = prec_interp[10 + 1 :]
+                ap_th = prec_trun.mean()
+                summary[cls_name] += [ap_th]
+
+            # AP Metric
+            ap = np.array(summary[cls_name][:num_ths]).mean()
+
+            # TP Error Metrics
+            tp_metrics = np.mean(cls_stats[:, num_ths : num_ths + 3], axis=0)
```

Comment:
what does num_ths + 3 represent? could that magic-number be made a module-level instance variable?

Reply:
`Pandas` could save the day here :smiley: (naming columns), but happy to make it some sort of length variable.